### ISSUE TITLE
publish 0.65 beta1 in the beta branch

### ISSUE THREAD

**Issue by kparal** [flathub/com.corsixth.corsixth#2]
There is a new CorsixTH 0.65 beta1:
https://groups.google.com/g/corsix-th-dev/c/re5wiPVAW7Y

Can you please publish it in the `beta` branch? It will then be accessible through the `flathub-beta` repository:
https://github.com/flathub/flathub#using-the-flathub-repository

Thanks!

**Comment by TheCycoONE** [comment]
@rmnvgr is this something you can take care of?

**Comment by rmnvgr** [comment]
@TheCycoONE I'm trying to build it, however I get an error when validating the Appstream metainfo:

```
AppData problem: tag-invalid : <release> versions are not in order [0.64 before 0.65-beta1]
```

However in the sources there is no `<release>` tag for 0.65-beta1, but it appears in the built files. Is it added during compilation? If yes, the most recent release must come first. And as it is a pre-release, it would be better if it was marked with the `type="development"` attribute.

**Comment by rmnvgr** [comment]
My bad, I was looking at `master` and apparently the `<release>` tag for 0.65-beta1 only exists in the `v0.65-beta1` tag. It is still in the incorrect order though.

**Comment by rmnvgr** [comment]
I backported https://github.com/CorsixTH/CorsixTH/commit/85d38c5f5ea0301218db5e7b31d9f4c859457205 to the 0.65-beta1 build, it is now building and will be available on the flathub beta channel soon.

**Comment by kparal** [comment]
@rmnvgr This can help:
https://github.com/Beep6581/RawTherapee/issues/5303
TLDR: Try using `v0.65~beta1` instead of `v0.65-beta1`.

If you still have issues, please publish your existing code in some named branch, so that we can have a look at it, perhaps I can help.

**Comment by kparal** [comment]
> I backported [CorsixTH/CorsixTH@85d38c5](https://github.com/CorsixTH/CorsixTH/commit/85d38c5f5ea0301218db5e7b31d9f4c859457205) to the 0.65-beta1 build, it is now building and will be available on the flathub beta channel soon.

This fixed the descending order of `<releases>`, as it should be, but it will still bite you once `0.65` is stable, I believe (the same issue as in the link above).

**Comment by rmnvgr** [comment]
> This fixed the descending order of `<releases>`, as it should be, but it will still bite you once `0.65` is stable, I believe (the same issue as in the link above).

I guess so, indeed! @TheCycoONE, is it possible to use `~` instead of `-` for pre-release versioning in the Appstream metainfo?

**Comment by TheCycoONE** [comment]
Sure, do you want to submit the PR?

**Comment by rmnvgr** [comment]
@TheCycoONE I will.

The beta is now available in the Flathub beta channel 🎉

**Comment by TheCycoONE** [comment]
FWIW I wasn't able to find any reference to the version format in the AppStream documentation; I'd expect it in e.g. https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-releases

**Comment by kparal** [comment]
I filed an RFE for it:
https://github.com/ximion/appstream/issues/317
